src/prefect/server/api/server.py

```diff
@@ -144,6 +157,15 @@ async def integrity_exception_handler(request: Request, exc: Exception):
     )
 
 
+async def retry_exception_handler(request: Request, exc: Exception):
+    """Return a 503 so the client can try again."""
+    return JSONResponse(
+        content={"exception_message": "Service Unavailable"},
+        status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+        headers={"Retry-After": str(get_retry_after_value())},
```

Comment:
Do we need a retry after header? This will prevent the client from doing an exponential backoff

Reply:
same as above 😅